src/main/java/io/github/fusionflux/portalcubed/content/prop/HammerItem.java

```diff
@@ -0,0 +1,26 @@
+package io.github.fusionflux.portalcubed.content.prop;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.Entity.RemovalReason;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.Level;
+
+public class HammerItem extends Item {
+	public HammerItem(Properties settings) {
+		super(settings);
+	}
+
+	public static void destroyProp(Player user, Level world, Prop prop) {
+		prop.remove(RemovalReason.KILLED);
+		var propItem = prop.getPickResult();
```

Comment:
this should be handled with loot tables.

Reply:
Why

Reply:
datapackability

Reply:
I don't feel like that is needed here

Reply:
I don't think loot tables can get NBT from the entity killed, so making props drop with their name and CMD should be done in code like it was pre-rewrite

Reply:
Looking into it more, this does actually seem possible with loot tables

Reply:
even if not possible with vanilla stuff, we can add a new function